### PR TITLE
FEAT: Adding money and smallmoney support in executemany

### PR DESCRIPTION
**Pull request by gargsaumya**
### Work Item / Issue Reference  

> AB#34936
AB#34937

> GitHub Issue: #<ISSUE_NUMBER>

-------------------------------------------------------------------
### Summary   
  

This pull request improves the handling of `decimal.Decimal` values and `None` (NULL) values when using `executemany` to insert data into MONEY and SMALLMONEY columns in SQL Server. It also adds comprehensive tests to verify correct roundtrip and NULL handling for these types.

Improvements to parameter processing in `executemany`:

* Updated the parameter processing logic in `cursor.py` to ensure that `decimal.Decimal` values are properly converted for VARCHAR columns and that `None` values are correctly skipped during conversion. This helps prevent type errors and ensures correct database insertion for MONEY, SMALLMONEY, and related types.

Expanded test coverage for MONEY and SMALLMONEY types:

* Added `test_money_smallmoney_roundtrip_executemany` to verify that inserting and retrieving MONEY and SMALLMONEY values using `executemany` with `decimal.Decimal` works as expected, including checks for value equality and type.
* Added `test_money_smallmoney_executemany_null_handling` to ensure that inserting NULLs into MONEY and SMALLMONEY columns via `executemany` behaves correctly and returns the expected results.

### CODE REVIEW

File: mssql_python/cursor.py [hunk 1]
```diff
@@ -1677,14 +1677,21 @@ def executemany(self, operation: str, seq_of_parameters: list) -> None:
...
+                # Existing numeric conversion
+                elif (parameters_type[i].paramSQLType in 
                     (ddbc_sql_const.SQL_DECIMAL.value, ddbc_sql_const.SQL_NUMERIC.value) and
-                    not isinstance(val, decimal.Decimal) and val is not None):
+                    not isinstance(val, decimal.Decimal)):
                     try:
                         processed_row[i] = decimal.Decimal(str(val))
                     except:
-                        pass  # Keep original value if conversion fails
+                        pass
```
Comment (sumitmsft) [inline mssql_python/cursor.py]: There is no logging or warning if conversion fails. Logging the conversion error would help troubleshooting.

**Review by sumitmsft — CHANGES_REQUESTED** [review]
Added a small comment..

File: tests/test_004_cursor.py [hunk 2]
```diff
@@ -6942,6 +6942,87 @@ def test_money_smallmoney_invalid_values(cursor, db_connection):
         drop_table_if_exists(cursor, "dbo.money_test")
         db_connection.commit()
 
+def test_money_smallmoney_roundtrip_executemany(cursor, db_connection):
```
Comment (sumitmsft) [inline tests/test_004_cursor.py:6944]: @gargsaumya Can you please also check for the following edge cases:

- Extremely large or small values: attempt to insert values at/near the boundaries of MONEY (-922337203685477.5808 to 922337203685477.5807) and SMALLMONEY (-214748.3648 to 214748.3647) ranges.
- Invalid type handling: check for incorrect types (e.g., strings, floats) being passed in executemany.
- Exception/Failure cases: ensure appropriate exceptions are raised for out-of-range, overflow, or invalid type cases for MONEY/SMALLMONEY.

Comment (gargsaumya) [reply tests/test_004_cursor.py:6944]: Some of these tests were already included in the previous PR, I’ve added the remaining ones that were missing.

**Review by bewithgaurav — APPROVED** [review]

**Review by sumitmsft — APPROVED** [review]
